src/python/TaskWorker/Actions/Recurring/TapeRecallManager.py

```diff
@@ -0,0 +1,239 @@
+2. for tasks in KILLRECALL delete rule and set to KILLED
+"""
+import logging
+import sys
+import os
+import time
+import datetime
+import copy
+
+from http.client import HTTPException
+from urllib.parse import urlencode
+
+from rucio.common.exception import RuleNotFound
+
+from ServerUtilities import MAX_DAYS_FOR_TAPERECALL, getTimeFromTaskname
+from RESTInteractions import CRABRest
+from RucioUtils import getNativeRucioClient
+from TaskWorker.MasterWorker import getRESTParams
+from TaskWorker.Worker import failTask
+from TaskWorker.Actions.Recurring.BaseRecurringAction import BaseRecurringAction
+
+class TapeRecallManager(BaseRecurringAction):
+    """ interface needed by the way TW deals with recurring actions
+    must have a class which inherit from BaseRecurringAction
+    which implements the _execute method (with the unused argument "task"...pff)
+    name of this class, this file and the recurring action in TW config list
+    must be the same """
+    pollingTime = 60*4  # unit=minutes. Runs every 4 hours
+    rucioClient = None
+    privilegedRucioClient = None
+    crabserver = None
+
+    def _execute(self, config, task):  # pylint: disable=unused-argument
+        """ this is what we do at every polling cycle """
+        self.config = config
+        # setup logger, crabserver client, rucio client
+        self.init()
+        # do the work
+        self.handleRecall()
+        self.handleKill()
+
+
+    def handleKill(self):
+        """ looks for tasks in KILLRECALL and deals with them """
+        status = 'KILLRECALL'
+        tasksToWorkOn = self.getTasks(status)
+
+        for aTask in tasksToWorkOn:
+            taskName = aTask['tm_taskname']
+            self.logger.info("Working on task %s", taskName)
+
+            reqId = aTask['tm_DDM_reqid']
+            if not reqId:
+                self.logger.debug("tm_DDM_reqid' is not defined for task %s, skipping it", taskName)
+                continue
+            msg = "Task points to Rucio RuleId:  %s " % reqId
+            self.logger.info(msg)
+
+            # delete rule and set task status to killed
+            self.logger.info('Will delete rule %s', reqId)
+            self.privilegedRucioClient.delete_replication_rule(reqId)
+            self.updateTaskStatus(taskName, 'KILLED')
+            # Clean up previous "dataset on tape" warnings
+            self.deleteWarnings(taskName)
+            self.logger.info("Done on this task")
+
+
+    def handleRecall(self):
+        """ looks for tasks in TAPERECALL and deals with them """
+        status = 'TAPERECALL'
+        tasksToWorkOn = self.getTasks(status)
+        for aTask in tasksToWorkOn:
+            taskName = aTask['tm_taskname']
+            self.logger.info("Working on task %s", taskName)
+            # 1.) check for "waited too long"
+            if (time.time() - getTimeFromTaskname(str(taskName))) > MAX_DAYS_FOR_TAPERECALL*24*60*60:
```

Comment:
Not directly relate to this line but you should update https://github.com/dmwm/CRABServer/blob/55f2ac9b2458dd8cb010812b431710e9918b1fc2/src/python/ServerUtilities.py#L83 to current policy (30 days) and also use it in
https://github.com/dmwm/CRABServer/blob/55f2ac9b2458dd8cb010812b431710e9918b1fc2/src/python/TaskWorker/Actions/DBSDataDiscovery.py#L193

Reply:
No. It is correct as it is. But clearly not explained well enough to be understandable.
Do you think that stating current policien in the comments [here](https://github.com/dmwm/CRABServer/blob/55f2ac9b2458dd8cb010812b431710e9918b1fc2/src/python/TaskWorker/Actions/DBSDataDiscovery.py#L93-L102) can do ? Alas policies have two components: the actions (in the code like `requestTapeRecall()` ) and the value of the constants (in `ServerUtilities.py`).

Reply:
No no. I am misunderstanding this. I remember this https://github.com/dmwm/CRABServer/pull/7591 but not https://github.com/dmwm/CRABServer/pull/7612 and that why I am confused.

I saw latest commit and it looks much better to understand.